### PR TITLE
Union of index objects

### PR DESCRIPTION
**Pull request by frankenjoe**
Adds `utils.union()` to create the union of a list of index objects.

### Example:

```python
objs = [
    audformat.segmented_index(['f1', 'f2'], [0, 0], [2, 2]),
    audformat.segmented_index(['f1', 'f2', 'f3'], [1, 0, 0], [2, 2, 2]),
    audformat.filewise_index(['f1', 'f1', 'f2', 'f3']),
]
audformat.utils.union(objs)
```
```
MultiIndex([('f1', '0 days 00:00:00', '0 days 00:00:02'),
            ('f1', '0 days 00:00:00',               NaT),
            ('f1', '0 days 00:00:01', '0 days 00:00:02'),
            ('f2', '0 days 00:00:00', '0 days 00:00:02'),
            ('f2', '0 days 00:00:00',               NaT),
            ('f3', '0 days 00:00:00', '0 days 00:00:02'),
            ('f3', '0 days 00:00:00',               NaT)],
           names=['file', 'start', 'end'])
```

### CODE REVIEW

**Comment by hagenw** [comment]
To be symmertical with `utils.intersect()` I would expect the following output of `audformat.utils.union(objs)`:

```
MultiIndex([('f1', '0 days 00:00:00',               NaT),
            ('f2', '0 days 00:00:00',               NaT),
            ('f3', '0 days 00:00:00',               NaT)],
           names=['file', 'start', 'end'])
```
but I agree that for most use cases your implementation makes a lot more sense.

**Comment by frankenjoe** [comment]
Yes, I would prefer to stick to the proposed implementation.

File: audformat/core/utils.py [hunk 1]
```diff
@@ -419,8 +419,10 @@ def to_segmented_index(
             :ref:`table specifications <data-tables:Tables>`
 
     """
-    if obj.empty or index_type(obj) == define.IndexType.SEGMENTED:
+    if index_type(obj) == define.IndexType.SEGMENTED:
         return obj
+    elif obj.empty:
+        return segmented_index()
```
Comment (hagenw) [inline audformat/core/utils.py:425]: Why this change to an existing function?

File: audformat/core/utils.py [hunk 2]
```diff
@@ -439,3 +441,87 @@ def to_segmented_index(
     obj = obj.reset_index(drop=True)
     obj.index = index
     return obj
+
+
+def union(
+    objs: typing.Sequence[typing.Union[pd.Index]],
```
Comment (hagenw) [inline audformat/core/utils.py:447]: Why Union?

Maybe:
```suggestion
    objs: typing.Sequence[pd.Index],
```

File: audformat/core/utils.py [hunk 3]
```diff
@@ -439,3 +441,87 @@ def to_segmented_index(
...
+                    ('f2', '0 days 00:00:00',               NaT),
+                    ('f2', '0 days 00:00:00', '0 days 00:00:01'),
+                    ('f3', '0 days 00:00:00',               NaT),
+                    ('f3', '0 days 00:00:00', '0 days 00:00:01'),
+                    ('f3', '0 days 00:00:01', '0 days 00:00:02'),
+                    ('f4', '0 days 00:00:00',               NaT),
+                    ('f4', '0 days 00:00:00', '0 days 00:00:01')],
+                   names=['file', 'start', 'end'])
+
+
```
Comment (hagenw) [inline audformat/core/utils.py:502]:
```suggestion
```

File: audformat/core/utils.py [hunk 4]
```diff
@@ -439,3 +441,87 @@ def to_segmented_index(
...
+    if index_type(index) == define.IndexType.SEGMENTED:
+        # asserts that start and end are of type 'timedelta64[ns]'
+        if index.empty:
+            index = segmented_index()
+        elif index.levels[2].empty:
+            index.set_levels(
+                [pd.to_timedelta([])],
+                level=[2],
+                inplace=True,
+            )
```
Comment (hagenw) [inline audformat/core/utils.py:525]: When can it happen that `index.levels[2].empty`?

Does it also mean if I pass two index objects with missing second level the returned index will then contain a second level?

File: audformat/core/utils.py (hunk 1 above)
Comment (frankenjoe) [reply audformat/core/utils.py:425]: This fixes a bug we had - before when you passed an empty `filewise` index it was returning it instead of an empty `segmented` index.

File: audformat/core/utils.py (hunk 4 above)
Comment (frankenjoe) [reply audformat/core/utils.py:525]: > When can it happen that index.levels[2].empty?

Yeah, took me a while to understand why we need this. The reason we have to add this is that `pandas` stores an empty array if all `end` values are set to `NaT`. And unfortunately `pd.Index.union()` converts from `timedelta` to `datetime` in that case. Don't ask me why they do it, but this fixes it.

> Does it also mean if I pass two index objects with missing second level the returned index will then contain a second level?

As I just said, the second level is not really missing, it's just containing only `pd.NaT` and in that case `pandas` stores and empty array. Probably for efficiency, but not sure.

File: audformat/core/utils.py (hunk 1 above)
Comment (hagenw) [reply audformat/core/utils.py:425]: OK, can you then also update the test for `to_segmented_index()` to cover this error.

File: audformat/core/utils.py (hunk 4 above)
Comment (hagenw) [reply audformat/core/utils.py:524]: OK, makes all sense, but please add a comment explaining why we do it to the code.

Comment (frankenjoe) [reply audformat/core/utils.py:527]: Honestly, the way pandas handles `timedelta` and `datetime` does not make much sense to me :)

I now added a comment explaining why we need this.

File: audformat/core/utils.py (hunk 1 above)
Comment (frankenjoe) [reply audformat/core/utils.py:425]: done

File: audformat/core/utils.py (hunk 4 above)
Comment (hagenw) [reply audformat/core/utils.py:527]: Yes, I also don't like how `pandas` handles it.
